app/src/org/commcare/activities/EntitySelectActivity.java

```diff
@@ -210,21 +217,15 @@ public void onChanged() {
     }
 
     private void setSearchBannerState() {
-        if (!"".equals(adapter.getSearchQuery())) {
+        if (!"".equals(adapter.getSearchQuery()) || adapter.isFilteringByCalloutResult()) {

```

Comment:
Not sure why this was set up so that the clear search button would not appear in the results banner for normal searches, but I can't see any reason why it should be that way. On newer devices, this isn't a big deal, because Android's SearchView has its own clear button. But on older devices that use our custom search view footer, the clear button on the search banner is the only place where it is possible to clear your search.